Automattic/wp-calypso
# webauthn: Client support behind a config flag

This pull request adds front-end support for using hardware tokens / webauthn as a second factor. 

### Changes proposed in this Pull Request

* Add a new section in `/me/security/two-step` where a user with a browser that supports webauthn can add new keys.
* Adds a new login block/action that handles the authentication part (second factor) when a given account has a registered hardware device.

### Testing instructions

Test live: https://calypso.live/?branch=add/gm19-webauthn
To add a new security key, one needs to setup first the standard two factor authentication. For now, this requires a proxied connection.

- Navigate to `/me/security/two-step`.
- Under the `Security Key` section, register a new key. Follow the instructions.
- Under a private window, log in to your test account. Make sure you are prompted to use the registered key. Click on the `Continue with security key` button and follow the instructions.

### Screenshots

#### Register key section
Before
<img width="733" alt="Screenshot 2019-09-20 at 16 12 51" src="https://user-images.githubusercontent.com/1081870/65334655-1fcac980-dbc3-11e9-868a-2a9ebd4add0a.png">
After
<img width="730" alt="Screenshot 2019-09-20 at 16 13 20" src="https://user-images.githubusercontent.com/1081870/65334661-25c0aa80-dbc3-11e9-9633-5839e7b2fbdd.png">

#### Login flow with a registered key
<img width="436" alt="Screenshot 2019-09-20 at 16 19 21" src="https://user-images.githubusercontent.com/1081870/65334682-32dd9980-dbc3-11e9-9021-8b23a6f0af61.png">
